src/framework/audio/internal/rpc/platform/general/generalrpcchannel.cpp

```diff
@@ -0,0 +1,210 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2025 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#include "generalrpcchannel.h"
+
+#include <thread>
+
+#include "../../../audiosanitizer.h"
+
+#include "log.h"
+
+//#define RPC_LOGGING_ENABLED
+
+#ifdef RPC_LOGGING_ENABLED
+#define RPCLOG() LOGDA()
+#else
+#define RPCLOG() LOGN()
+#endif
+
+using namespace muse::audio::rpc;
+
+void GeneralRpcChannel::initOnWorker()
+{
+    ONLY_AUDIO_WORKER_THREAD;
+    m_workerThreadID = std::this_thread::get_id();
+}
+
+bool GeneralRpcChannel::isWorkerThread() const
+{
+    return std::this_thread::get_id() == m_workerThreadID;
+}
+
+void GeneralRpcChannel::process()
+{
+    if (isWorkerThread()) {
+        receive(m_mainRpcData, m_workerRpcData);
+    } else {
+        receive(m_workerRpcData, m_mainRpcData);
+    }
+}
+
+void GeneralRpcChannel::receive(RpcData& from, RpcData& to) const
+{
+    MsgQueue msgQueue;
+    StreamMsgQueue streamMsgQueue;
+    {
+        std::scoped_lock<std::mutex> lock(from.mutex);
+        msgQueue.swap(from.queue);
+        streamMsgQueue.swap(from.streamQueue);
+    }
+
+    // msgs
+    while (!msgQueue.empty()) {
+        const Msg& m = msgQueue.front();
+
+        RPCLOG() << "callId: " << m.callId
```

Comment:
I suggest hiding this code using an ifdef. To avoid even the slightest chance that this code will do something in release builds. Because right now, for example, it creates 2 strings